underfs/s3/src/main/java/tachyon/underfs/s3/S3OutputStream.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.underfs.s3;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.UUID;
+
+import org.jets3t.service.S3Service;
+import org.jets3t.service.ServiceException;
+import org.jets3t.service.model.S3Object;
+import org.jets3t.service.utils.Mimetypes;
+
+import tachyon.util.CommonUtils;
+
+public class S3OutputStream extends OutputStream {
+  private final String mBucketName;
+  private final String mKey;
+  private final OutputStream mLocalOutputStream;
+  private final File mFile;
+  private final S3Service mClient;
+
+  public S3OutputStream(String bucketName, String key, S3Service client) throws IOException {
+    mBucketName = bucketName;
```

Comment:
NIT: is bucketName eligible to be empty? if no, probably have a checkArgument here?